### ISSUE TITLE
Add the ability to produce visualization data given a list of `end times.`

### ISSUE THREAD

**Issue by hlokavarapu** [geodynamics/aspect#1702]
If one is interested in producing output results at specific times, per say, `t = 0.05, 0.10, 0.15` non-dim time, the only possibility is to set off 3 sets of runs in a continuous fashion. For example, the first run will compute from `t=0` to `t=0.5`, generate a checkpoint file and visualization data, resume the computation at `t=0.5` and run out to `t=0.10`, generate a checkpoint file and produce another visualization data set, and then repeat once more to hit the end time of `t=0.15`. The result is that there are three visualization outputs at exactly `t = 0.05, 0.10, 0.15`.

**Comment by hlokavarapu** [comment]
In order to implement this functionality, is it best to make this visible as a parameter of the termination criteria?

**Comment by bangerth** [comment]
No, I think it should go through the same code path that currently deals with the end time. Maybe have a vector of end times and always try to hit the next one we haven't been to yet?
